test/framework/clusterctl/e2e_config.go

```diff
@@ -245,6 +248,90 @@ type Files struct {
 	TargetName string `json:"targetName,omitempty"`
 }
 
+// ResolveReleases converts release markers to release version.
+func (c *E2EConfig) ResolveReleases() error {
+	scheme, host, err := goproxy.GetSchemeAndHost(os.Getenv("GOPROXY"))
+	if err != nil {
+		return err
+	}
+	if scheme == "" || host == "" {
+		return fmt.Errorf("releasemarker does not support disabling the go proxy: GOPROXY=%q", os.Getenv("GOPROXY"))
+	}
+	goproxyClient := goproxy.NewClient(scheme, host)
+	for i := range c.Providers {
+		provider := &c.Providers[i]
+		for j := range provider.Versions {
+			version := &provider.Versions[j]
+			if version.Type != URLSource {
+				continue
+			}
+			// Skipping versions that are not a resolvable marker. Resolvable markers are surrounded by `{}`
+			if !strings.HasPrefix(version.Name, "{") || !strings.HasSuffix(version.Name, "}") {
+				continue
+			}
+			releaseMarker := strings.TrimLeft(strings.TrimRight(version.Name, "}"), "{")
+			ver, err := resolveReleaseMarker(releaseMarker, goproxyClient)
+			if err != nil {
+				return errors.Wrapf(err, "Failed resolving release url %q", version.Name)
+			}
+			ver = "v" + ver
+			version.Value = strings.Replace(version.Value, version.Name, ver, 1)
+			version.Name = ver
+		}
+	}
+	return nil
+}
+
+// resolveReleaseMarker resolves releaseMarker string to verion string e.g.
+// - Resolves "go://sigs.k8s.io/cluster-api@v1.0" to the latest stable patch release of v1.0.
+// - Resolves "go://sigs.k8s.io/cluster-api@latest-v1.0" to the latest patch release of v.1.0 including rc and pre releases.
```

Comment:
I am not sure why we should think of a release with `latest-v1.0` version, it does not exist at all, is not it?

Reply:
It is just an example of how this function will resolve release markers, this function will return the latest release even if they are `rc` or `patch`  against `latest-vX.0`. In the case of `latest-v1.0` function will return `v1.0.5` because that is the latest release of `v1.0`, `rc` and `patch` releases for `v1.0` are not the latest. I can change it to some other release if this does not seem right. I can also change it to `latest-vX.0`.

Reply:
@adilGhaffarDev thanks for explanation, I understood it differently at first, all good as is from my side!